### PR TITLE
Change Task[Any] to Task[_T] in return statements

### LINKED ISSUE #3803: asyncio.create_task returns a Task of type Any
https://github.com/python/typeshed/blob/master/stdlib/3/asyncio/tasks.pyi#L126

The type annotation for `create_task` is:
```
        def create_task(
            coro: Union[Generator[Any, None, _T], Awaitable[_T]], *, name: Optional[str] = ...,
        ) -> Task[Any]: ...
```
Why not annotate the return as `Task[_T]`? It appears to me that the `Task` class supports this and it would be more correct. But given that this wasn't done in the first place makes me wonder perhaps there is a reason this doesn't work?

There's a couple other instances of this, in particular in the `Task` class itself

### PR DESCRIPTION
**Pull request by lladhibhutall**
closes #3803

### CODE REVIEW

File: stdlib/3/asyncio/tasks.pyi [hunk 1]
```diff
@@ -93,9 +93,9 @@ def wait_for(fut: _FutureT[_T], timeout: Optional[float],
 
 class Task(Future[_T], Generic[_T]):
     @classmethod
-    def current_task(cls, loop: Optional[AbstractEventLoop] = ...) -> Task[Any]: ...
+    def current_task(cls, loop: Optional[AbstractEventLoop] = ...) -> Task[_T]: ...
```
Comment (CraftSpider) [inline stdlib/3/asyncio/tasks.pyi]: I don't think it makes sense to change it to `_T` on the methods without `_T` parameters. `_T` isn't bound outside a class, or on the class methods. Also, all_tasks and current_task aren't guaranteed to return any specific type

Comment (lladhibhutall) [reply stdlib/3/asyncio/tasks.pyi]: @CraftSpider Thanks for the review, I am just curious how did you find out that `all_tasks` and `current_task` aren't guaranteed to return any specific type. I am a beginner.

Comment (lladhibhutall) [reply stdlib/3/asyncio/tasks.pyi]: @CraftSpider Thanks for the review, I am just curious how did you find out that all_tasks and current_task aren't guaranteed to return any specific type. I am a beginner.

Comment (jwiesmore) [reply stdlib/3/asyncio/tasks.pyi]: They do return `Task`, by 'any type' I mean `Task[Any]` not `Task[_T]`. They return either the current task or all unfinished tasks. Personally, I've used asyncio and just remembered; but for the actual explanation, https://docs.python.org/3/library/asyncio-task.html#introspection is the documentation for them.

Comment (CraftSpider) [reply stdlib/3/asyncio/tasks.pyi]: They do return `Task`, by 'any type' I mean `Task[Any]` not `Task[_T]`. They return either the current task or all unfinished tasks. Personally, I've used asyncio and just remembered; but for the actual explanation, https://docs.python.org/3/library/asyncio-task.html#introspection is the documentation for them.

Comment (lladhibhutall) [reply stdlib/3/asyncio/tasks.pyi]: @CraftSpider Yeah should have checked the docs, `current_task` can also return `None`, so put an `Optional`, thanks.

**Pull request merged**